### PR TITLE
changed component source link generation

### PR DESCRIPTION
**Pull request by Villhellm**
**Description:**
Integration docs pages such as `sensor.mqtt` link the source page to `components/sensor` instead of `components/mqtt`. This can be misleading for people that are looking for the source of the specific component they are looking at.

**Pull request in home-assistant (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next Home Assistant release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

**Review by frenck — APPROVED** [review]
Nice! Thanks!

Please adjust the target branch to `current` and rebase if needed. This change is not depended on an Home Assistant release and should therefor not go into `next` but `current`.

**Review by frenck — CHANGES_REQUESTED** [review]
Wrong button, let me try again.

**Comment by Villhellm** [comment]
Oops, sorry about that. I thought that I had. Thanks for all the hard work you've been doing this Hacktober, frenck. We all really appreciate it!

**Comment by frenck** [comment]
Your welcome @Villhellm! However, I think the big thanks should go to all the contributors and people participating in Hacktoberfest. Those are the ones who deserve a 🎖 

PS: You need to rebase this PR. Feel free to ping me if you need help with that.

**Comment by Villhellm** [comment]
Well, looks like I screwed that one up :( should I just make a new PR on the proper branch @frenck ?
